redis-om-spring/src/main/java/com/redis/om/spring/ops/json/JSONOperationsImpl.java

```diff
@@ -1,153 +1,149 @@
-import java.util.Arrays;
-import java.util.List;
-
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
 import com.redis.om.spring.client.RedisModulesClient;
-import com.redis.om.spring.ops.Command;
-import com.redislabs.modules.rejson.JReJSON.ExistenceModifier;
-import com.redislabs.modules.rejson.Path;
+import redis.clients.jedis.json.JsonSetParams;
+import redis.clients.jedis.json.Path;
+import redis.clients.jedis.json.Path2;
 
-import redis.clients.jedis.Jedis;
-import redis.clients.jedis.Protocol;
-import redis.clients.jedis.util.SafeEncoder;
+import java.util.Arrays;
+import java.util.List;
 
 public class JSONOperationsImpl<K> implements JSONOperations<K> {
 
-  RedisModulesClient client;
+  private final Gson gson;
+  final RedisModulesClient client;
 
-  public JSONOperationsImpl(RedisModulesClient client) {
+  public JSONOperationsImpl(RedisModulesClient client, GsonBuilder gson) {
     this.client = client;
+    this.gson = gson.create();
   }
 
   @Override
   public Long del(K key, Path path) {
-    return client.clientForJSON().del(key.toString(), path);
+    return client.clientForJSON().jsonDel(key.toString(), path);
   }
 
   @Override
-  public <T> T get(K key) {
-    return client.clientForJSON().get(key.toString());
+  public <T> T get(K key, Class<T> clazz) {
+    return gson.fromJson(client.clientForJSON().jsonGetAsPlainString(key.toString(), Path.ROOT_PATH), clazz);
   }
 
   @Override
-  public <T> T get(K key, Class<T> clazz, Path... paths) {
-    return client.clientForJSON().get(key.toString(), clazz, paths);
+  public <T> T get(K key, Class<T> clazz, Path path) {
+    return gson.fromJson(client.clientForJSON().jsonGetAsPlainString(key.toString(), path), clazz);
   }
 
-  @Override
-  public <T> List<T> mget(Class<T> clazz, @SuppressWarnings("unchecked") K... keys) {
-    String[] keysAsStrings = Arrays.asList(keys).stream().map(Object::toString).toArray(String[]::new);
-    return client.clientForJSON().mget(clazz, keysAsStrings);
+  @SafeVarargs @Override
+  public final <T> List<T> mget(Class<T> clazz, K... keys) {
+    String[] keysAsStrings = Arrays.stream(keys).map(Object::toString).toArray(String[]::new);
+    return client.clientForJSON().jsonMGet(keysAsStrings)
+        .stream().map(jsonArr -> jsonArr.get(0))
+        .map(Object::toString)
+        .map(str -> gson.fromJson(str, clazz))
+        .toList();
   }
 
-  @Override
-  public <T> List<T> mget(Path path, Class<T> clazz, @SuppressWarnings("unchecked") K... keys) {
-    String[] keysAsStrings = Arrays.asList(keys).stream().map(Object::toString).toArray(String[]::new);
-    return client.clientForJSON().mget(path, clazz, keysAsStrings);
+  @SafeVarargs @Override
+  public final <T> List<T> mget(Path2 path, Class<T> clazz, K... keys) {
+    String[] keysAsStrings = Arrays.stream(keys).map(Object::toString).toArray(String[]::new);
+
+    return client.clientForJSON().jsonMGet(path, keysAsStrings)
+        .stream()
```

Comment:
Do we need to take the first element because it's an array of JSONArray?

Reply:
Not on this case